### PR TITLE
build: Add meson build system

### PR DESCRIPTION
**Pull request by jtojnar**
https://wiki.gnome.org/Initiatives/GnomeGoals/MesonPorting

### CODE REVIEW

**Comment by Kaffeine** [comment]
:open_mouth: . I'm sorry I know nothing about Meson and maintain mostly Qt parts. If this works for you and there is no objection from anyone then I'm agree to merge it. :confused:

File: .gitignore [hunk 1]
```diff
@@ -48,6 +48,7 @@ Makefile.in
 data/org.freedesktop.Telepathy.ConnectionManager.idle.service
 
 extensions/_gen
```
Comment (jtojnar) [inline .gitignore]: What I do not like is that the extensions are generated inside the build tree.

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+#   make the release, tag it
+#   set nano_version to 1
+
+project(
+	'telepathy-idle',
+	'c',
+	version: '0.2.0.1',
+	default_options: [
+		'c_std=c99',
+		# 'werror=@0@'.format((meson.project_version().split('.')[3] == '0').to_string()), # development version
```
Comment (jtojnar) [inline meson.build]: This will need to be removed: https://github.com/mesonbuild/meson/issues/4577#issuecomment-470117483

**Comment by okias** [comment]
@jtojnar @Kaffeine  can be merged then? Meson would be nice

**Comment by Kaffeine** [comment]
@jtojnar please remove the WIP mark from the PR title if you're OK with this being merged.

I would be happy to merge this as soon as the author agree, I'll be unavailable until November 5.
It would be fantastic to eventually have meson build for telepathy-mission-control and telepathy-glib too.

File: .gitignore (hunk 1 above)
Comment (jtojnar) [reply .gitignore]: Actually, they are not with meson, so everything is fine.

**Comment by jtojnar** [comment]
I dropped https://github.com/TelepathyIM/telepathy-idle/pull/4#discussion_r265544907, switched to finding Python using the [python module](https://mesonbuild.com/Python-module.html) (requires bumping minimal Meson version), moved twisted test to a separate suite so we can easily disable it, and fixed formatting.

**Comment by Kaffeine** [comment]
Merged as 108e30c59bf2c7f96611129c67d060f232ddbaf4.
Thank you!

**Comment by Kaffeine** [comment]
I am sorry, I messed up with your branch. You targeted `master` before Python 3 support landed and I merged the code to `telepathy-idle-0.2` after the code switched to P3. :facepalm:.
I fixed the Meson build in the 0.2.2 release (just aired).

**Pull request closed without merging**